### PR TITLE
Refine regression test  with more clear messages and logs

### PR DESCRIPTION
**Pull request by freewheel70**
[AB#572575](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/572575)

Refine regression test  with more clear messages and logs for better debug experience.

### CODE REVIEW

File: test/docfx.RegressionTest/RegressionTestResult.cs [hunk 1]
```diff
@@ -20,4 +20,12 @@ internal class RegressionTestResult
     public string? CrashMessage { get; set; }
 
     public int MoreLines { get; set; }
+
+    public override string ToString() => $"{{{nameof(Succeeded)}={Succeeded}, " +
```
Comment (yufeih) [inline test/docfx.RegressionTest/RegressionTestResult.cs:24]: I don't quite like this manual `ToString` implementation. Does changing this to a `record` automatically gives you the desired `ToString` behavior? Or is it sufficient to dump the object out as JSON?

File: test/docfx.RegressionTest/RegressionTest.cs [hunk 2]
```diff
@@ -174,7 +174,9 @@ private static bool Test(Options opts, string workingFolder)
 
         Compare(testResult, opts, workingFolder, outputPath, baseLinePath);
         Console.BackgroundColor = ConsoleColor.DarkMagenta;
-        Console.WriteLine($"Test Pass {workingFolder}");
+        Console.WriteLine($"Test {(testResult.Succeeded ? "Pass" : "Fail")} {workingFolder}");
+        Console.WriteLine("Test Result Summary:");
+        Console.WriteLine(testResult.ToString());
```
Comment (herohua) [inline test/docfx.RegressionTest/RegressionTest.cs:179]: nit: `ToString()` here is not necessary

Comment (freewheel70) [reply test/docfx.RegressionTest/RegressionTest.cs:179]: Thanks for review, I will update the code.

File: test/docfx.RegressionTest/RegressionTestResult.cs (hunk 1 above)
Comment (freewheel70) [reply test/docfx.RegressionTest/RegressionTestResult.cs:24]: @herohua @yufeih 
Thanks for your review. 
Actually in the ToString implementation, I ignore some fields (i.e. HotMethods), and customize the `Diff` field output. So changing to `record` may be not help.
But after a second thought, I guess it is not necessary add ToString() method here, it is better to put the output logic directly in Console.WriteLine() statement.

Comment (yufeih) [reply test/docfx.RegressionTest/RegressionTestResult.cs:24]: We could change the `HotMethods` to an array and auto print it here as well. This is only for our own console logging and no need to get fancy.

Comment (freewheel70) [reply test/docfx.RegressionTest/RegressionTestResult.cs]: HotMethods  is printed out at https://github.com/dotnet/docfx/blob/5952ea7b299a78d5b74190ac8d83db546757b596/test/docfx.RegressionTest/RegressionTest.cs#L285

**Comment by herohua** [comment]
<details><summary>🧭<a href='https://github.com/MicrosoftDocs/PowerShell-Docs'>PowerShell-Docs</a>(84.1413445s | exceed 75s, 3978MB)</summary>



</details>

**Comment by herohua** [comment]
<details><summary>⚠<a href='https://github.com/MicrosoftDocs/sql-docs-pr'>sql-docs-pr</a>(57.85727s, 5738MB)</summary>

```diff
diff --git "a/D:/docfx.TestData\\regression-test.sql-docs-pr\\output/.errors.log" "b/D:/docfx.TestData\\regression-test.sql-docs-pr\\.temp/.errors.log"
index c3ad6fc..3ec1788 100644
--- "a/D:/docfx.TestData\\regression-test.sql-docs-pr\\output/.errors.log"
+++ "b/D:/docfx.TestData\\regression-test.sql-docs-pr\\.temp/.errors.log"
@@ -591325,3 +591325,20 @@
   "ms.topic": "conceptual",
   "document_url": "https://review.docs.microsoft.com/en-us/help/contribute/validation-ref/ms-author-invalid?branch=main"
 }
+{
+  "message_severity": "warning",
+  "code": "ms-author-invalid",
+  "message": "Invalid value for 'ms.author', 'anrampal' is not a valid Microsoft alias.",
+  "file": "docs/tools/synapse-pathway/synapse-pathway-download.md",
+  "line": 14,
+  "end_line": 14,
+  "column": 12,
+  "end_column": 20,
+  "log_item_type": "user",
+  "property_path": "ms.author",
+  "ms.author": "anrampal",
+  "ms.prod": "sql",
+  "ms.technology": "tools-other",
+  "ms.topic": "conceptual",
+  "document_url": "https://review.docs.microsoft.com/en-us/help/contribute/validation-ref/ms-author-invalid?branch=main"
+}


```

</details>

**Comment by herohua** [comment]
<details><summary>🧭<a href='https://github.com/MicrosoftDocs/azure-docs-pr'>azure-docs-pr</a>(115.7152764s | exceed 100s, 8876MB)</summary>

```diff
diff --git "a/D:/docfx.TestData\\regression-test.azure-docs-pr\\output/.publish.json" "b/D:/docfx.TestData\\regression-test.azure-docs-pr\\.temp/.publish.json"
index 9624478..b4b6a9e 100644
--- "a/D:/docfx.TestData\\regression-test.azure-docs-pr\\output/.publish.json"
+++ "b/D:/docfx.TestData\\regression-test.azure-docs-pr\\.temp/.publish.json"
@@ -801221,8 +801221,7 @@ IMPORTANT: This content is for the older Azure AD v1.0 endpoint, use the Microso
         "mikefrobbins",
         "kraigb",
         "DCtheGeek",
-        "Hazhzeng",
-        "gvashishtha"
+        "Hazhzeng"
       ],
       "updated_at": "2021-09-21 12:35 AM",
       "content_git_url": "https://github.com/MicrosoftDocs/azure-docs/blob/main/articles/azure-functions/machine-learning-pytorch.md",
diff --git "a/D:/docfx.TestData\\regression-test.azure-docs-pr\\output/azure/azure-functions/machine-learning-pytorch.mta.json" "b/D:/docfx.TestData\\regression-test.azure-docs-pr\\.temp/azure/azure-functions/machine-learning-pytorch.mta.json"
index 93a62c0..d2bf30b 100644
--- "a/D:/docfx.TestData\\regression-test.azure-docs-pr\\output/azure/azure-functions/machine-learning-pytorch.mta.json"
+++ "b/D:/docfx.TestData\\regression-test.azure-docs-pr\\.temp/azure/azure-functions/machine-learning-pytorch.mta.json"
@@ -14,8 +14,7 @@
     "mikefrobbins",
     "kraigb",
     "DCtheGeek",
-    "Hazhzeng",
-    "gvashishtha"
+    "Hazhzeng"
   ],
   "is_dynamic_rendering": true,
   "layout": "Conceptual",
diff --git "a/D:/docfx.TestData\\regression-test.azure-docs-pr\\output/azure/azure-functions/machine-learning-pytorch.raw.page.json" "b/D:/docfx.TestData\\regression-test.azure-docs-pr\\.temp/azure/azure-functions/machine-learning-pytorch.raw.page.json"
index 983e834..8d568c3 100644
--- "a/D:/docfx.TestData\\regression-test.azure-docs-pr\\output/azure/azure-functions/machine-learning-pytorch.raw.page.json"
+++ "b/D:/docfx.TestData\\regression-test.azure-docs-pr\\.temp/azure/azure-functions/machine-learning-pytorch.raw.page.json"
@@ -311,12 +311,6 @@ from .predict import predict_image_from_url⬇
           "profile_url": "https://github.com/Hazhzeng",
           "display_name": "RogerZeng",
           "id": "48038149"
-        },
-        {
-          "name": "gvashishtha",
-          "profile_url": "https://github.com/gvashishtha",
-          "display_name": "Gopal Vashishtha",
-          "id": "15335863"
         }
       ],
       "author": {
@@ -332,8 +326,7 @@ from .predict import predict_image_from_url⬇
       "mikefrobbins",
       "kraigb",
       "DCtheGeek",
-      "Hazhzeng",
-      "gvashishtha"
+      "Hazhzeng"
     ],
     "updated_at": "2021-09-21 12:35 AM",
     "content_git_url": "https://github.com/MicrosoftDocs/azure-docs/blob/main/articles/azure-functions/machine-learning-pytorch.md",
diff --git "a/D:/docfx.TestData\\regression-test.azure-docs-pr\\output/azure/batch/scripts/batch-cli-sample-add-application.raw.page.json" "b/D:/docfx.TestData\\regression-test.azure-docs-pr\\.temp/azure/batch/scripts/batch-cli-sample-add-application.raw.page.json"
index 117bf6c..dd3d19f 100644
--- "a/D:/docfx.TestData\\regression-test.azure-docs-pr\\output/azure/batch/scripts/batch-cli-sample-add-application.raw.page.json"
+++ "b/D:/docfx.TestData\\regression-test.azure-docs-pr\\.temp/azure/batch/scripts/batch-cli-sample-add-application.raw.page.json"
@@ -221,7 +221,7 @@ Each command in the table links to command-specific documentation.</p>⬇
         {
           "name": "annatisch",
           "profile_url": "https://github.com/annatisch",
-          "display_name": "annatisch",
+          "display_name": "Anna Tisch",
           "id": "8689453"
         }
       ],
diff --git "a/D:/docfx.TestData\\regression-test.azure-docs-pr\\output/azure/batch/scripts/batch-cli-sample-create-account.raw.page.json" "b/D:/docfx.TestData\\regression-test.azure-docs-pr\\.temp/azure/batch/scripts/batch-cli-sample-create-account.raw.page.json"
index 54f81f0..1b9ff09 100644
--- "a/D:/docfx.TestData\\regression-test.azure-docs-pr\\output/azure/batch/scripts/batch-cli-sample-create-account.raw.page.json"
+++ "b/D:/docfx.TestData\\regression-test.azure-docs-pr\\.temp/azure/batch/scripts/batch-cli-sample-create-account.raw.page.json"
@@ -208,7 +208,7 @@ resource group and all resources associated with it.</p>⬇
         {
           "name": "annatisch",
           "profile_url": "https://github.com/annatisch",
-          "display_name": "annatisch",
+          "display_name": "Anna Tisch",
           "id": "8689453"
         }
       ],
diff --git "a/D:/docfx.TestData\\regression-test.azure-docs-pr\\output/azure/batch/scripts/batch-cli-sample-run-job.raw.page.json" "b/D:/docfx.TestData\\regression-test.azure-docs-pr\\.temp/azure/batch/scripts/batch-cli-sample-run-job.raw.page.json"
index cdedb4a..c4d8679 100644
--- "a/D:/docfx.TestData\\regression-test.azure-docs-pr\\output/azure/batch/scripts/batch-cli-sample-run-job.raw.page.json"
+++ "b/D:/docfx.TestData\\regression-test.azure-docs-pr\\.temp/azure/batch/scripts/batch-cli-sample-run-job.raw.page.json"
@@ -247,7 +247,7 @@ resource group and all resources associated with it.</p>⬇
         {
           "name": "annatisch",
           "profile_url": "https://github.com/annatisch",
-          "display_name": "annatisch",
+          "display_name": "Anna Tisch",
           "id": "8689453"
         }
       ],
diff --git "a/D:/docfx.TestData\\regression-test.azure-docs-pr\\output/azure/virtual-desktop/install-office-on-wvd-master-image.raw.page.json" "b/D:/docfx.TestData\\regression-test.azure-docs-pr\\.temp/azure/virtual-desktop/install-office-on-wvd-master-image.raw.page.json"
index b98c334..5a43fd8 100644
--- "a/D:/docfx.TestData\\regression-test.azure-docs-pr\\output/azure/virtual-desktop/install-office-on-wvd-master-image.raw.page.json"
+++ "b/D:/docfx.TestData\\regression-test.azure-docs-pr\\.temp/azure/virtual-desktop/install-office-on-wvd-master-image.raw.page.json"
@@ -235,7 +235,7 @@ reg add HKLM\\SOFTWARE\\Policies\\Microsoft\\office\\16.0\\common\\officeupdate
         {
           "name": "DHB-MSFT",
           "profile_url": "https://github.com/DHB-MSFT",
-          "display_name": "Daniel H. Brown",
+          "display_name": "Dan Brown",
           "id": "32883970"
         },
         {


``````csharp
 29% | docfx!Microsoft.Docs.Build.ParallelUtility+<>c__DisplayClass2_0`1[System.__Canon].<ForEach>b__0(!0)
 25% | docfx!Microsoft.Docs.Build.DocsetBuilder+<>c__DisplayClass41_1.<Build>b__15(class Microsoft.Docs.Build.FilePath)
 23% | docfx!Microsoft.Docs.Build.PageBuilder.Build(class Microsoft.Docs.Build.ErrorBuilder,class Microsoft.Docs.Build.FilePath)
 19% | Microsoft.Docs.MarkdigExtensions!Microsoft.Docs.MarkdigExtensions.ResolveLinkExtension.UpdateLinks(class Markdig.Syntax.MarkdownObject)
 16% | docfx!Microsoft.Docs.Build.PageBuilder.LoadMarkdown(class Microsoft.Docs.Build.ErrorBuilder,class Microsoft.Docs.Build.FilePath)
 14% | docfx!Microsoft.Docs.Build.MarkdownEngine.ToHtml(class Microsoft.Docs.Build.ErrorBuilder,class System.String,class Microsoft.Docs.Build.SourceInfo,value class Microsoft.Docs.Build.MarkdownPipelineType,class Microsoft.Docs.Build.ConceptualModel,bool)
 13% | Markdig!Markdig.Markdown.ToHtml(class System.String,class Markdig.MarkdownPipeline)
 10% | Markdig!Markdig.Parsers.MarkdownParser.Parse()
  8% | Markdig!Markdig.Renderers.RendererBase.Write(class Markdig.Syntax.MarkdownObject)
  6% | docfx!Microsoft.Docs.Build.MarkdigUtility.<Replace>g__ReplaceCore|7_0(class Markdig.Syntax.MarkdownObject,class System.Func`2<class Markdig.Syntax.MarkdownObject,class Markdig.Syntax.MarkdownObject>)
  6% | docfx!Microsoft.Docs.Build.Docfx+<>c__DisplayClass6_0.<CreateCommand>b__0(class Microsoft.Docs.Build.CommandLineOptions)
  6% | docfx!Microsoft.Docs.Build.Docfx+<>c__DisplayClass4_0.<BuildCommand>b__0(class Microsoft.Docs.Build.CommandLineOptions)
  6% | docfx!Microsoft.Docs.Build.Builder.Run(class Microsoft.Docs.Build.CommandLineOptions,class Microsoft.Docs.Build.Package)
  6% | docfx!Microsoft.Docs.Build.Builder.Build(class Microsoft.Docs.Build.ErrorBuilder,class System.IProgress`1<class System.String>,class System.String[])
  4% | docfx!Microsoft.Docs.Build.MarkdigUtility.Visit(class Markdig.Syntax.MarkdownObject,class System.Func`2<class Markdig.Syntax.MarkdownObject,bool>)
  4% | docfx!Microsoft.Docs.Build.Builder+<>c__DisplayClass8_0.<Build>b__0(class Microsoft.Docs.Build.DocsetBuilder)
  4% | docfx!Microsoft.Docs.Build.PageBuilder.CreatePageOutput(class Microsoft.Docs.Build.ErrorBuilder,class Microsoft.Docs.Build.FilePath,class Newtonsoft.Json.Linq.JObject)
  4% | docfx!Microsoft.Docs.Build.MarkdownEngine.GetLink(class Microsoft.Docs.Build.LinkInfo)
  3% | docfx!Microsoft.Docs.Build.DocsetBuilder.Build(class System.String[])
  3% | docfx!Microsoft.Docs.Build.ParallelUtility.ForEach(class Microsoft.Docs.Build.LogScope,class Microsoft.Docs.Build.ErrorBuilder,class System.Collections.Generic.IEnumerable`1<!!0>,class System.Action`1<!!0>)
  3% | docfx!Microsoft.Docs.Build.LinkResolver.ResolveLink(value class Microsoft.Docs.Build.SourceInfo`1<class System.String>,class Microsoft.Docs.Build.FilePath,class Microsoft.Docs.Build.FilePath,class Microsoft.Docs.Validation.LinkNode,class System.String)
  3% | Markdig!Markdig.Renderers.RendererBase.WriteChildren(class Markdig.Syntax.ContainerBlock)
  3% | docfx!Microsoft.Docs.Build.Watch`1[System.__Canon].get_Value()
  3% | docfx!Microsoft.Docs.Build.DocsetBuilder.BuildFile(class Microsoft.Docs.Build.FilePath,class Microsoft.Docs.Build.ContentValidator,class Microsoft.Docs.Build.ResourceBuilder,class Microsoft.Docs.Build.PageBuilder,class Microsoft.Docs.Build.TocBuilder,class Microsoft.Docs.Build.RedirectionBuilder)
  3% | docfx!Microsoft.Docs.Build.LinkResolver.TryResolveAbsoluteLink(value class Microsoft.Docs.Build.SourceInfo`1<class System.String>,class Microsoft.Docs.Build.FilePath,class Microsoft.Docs.Build.FilePath,class Microsoft.Docs.Validation.LinkNode,class System.String)
  3% | docfx!Microsoft.Docs.Build.MarkdownEngine.GetImageLink(class System.String,class Markdig.Syntax.MarkdownObject,class System.String,class System.String)
  2% | Markdig!Markdig.Renderers.TextRendererBase.Render(class Markdig.Syntax.MarkdownObject)
  2% | docfx!Microsoft.Docs.Build.ErrorBuilder.AddRange(class System.Collections.Generic.IEnumerable`1<class Microsoft.Docs.Build.Error>)
  2% | docfx!Microsoft.Docs.Build.ContentValidator.ValidateLink(class Microsoft.Docs.Build.FilePath,class Microsoft.Docs.Validation.LinkNode)
  2% | docfx!Microsoft.Docs.Build.PageBuilder.CreateSystemMetadata(class Microsoft.Docs.Build.ErrorBuilder,class Microsoft.Docs.Build.FilePath,class Microsoft.Docs.Build.UserMetadata)

```

</details>

**Review by 928PJY — APPROVED** [review]

File: test/docfx.RegressionTest/RegressionTest.cs [hunk 3]
```diff
@@ -317,21 +319,30 @@ private static void Compare(RegressionTestResult testResult, Options opts, strin
...
+                MarkTaskFailed("Test failed, see the logs under /Summary/Build artifacts for details");
             }
 
-            Console.WriteLine($"##vso[artifact.upload artifactname=diff;]{diffFile}");
-            Console.WriteLine($"##vso[task.complete result=Failed]Test failed, see the logs under /Summary/Build artifacts for details");
+            var isTimeout = testResult.BuildTime.TotalSeconds > opts.Timeout;
+            if (isTimeout)
+            {
+                MarkTaskFailed($"Test failed, build timeout. Repo: {s_testName}; Expected Runtime: {opts.Timeout}s");
+                Console.WriteLine($"Test failed, build timeout. Repo: {s_testName}; Expected Runtime: {opts.Timeout}s");
```
[the remark this answers is not in the document]
Comment (freewheel70) [reply test/docfx.RegressionTest/RegressionTest.cs:342]: yeah, but I see another place to call `MarkTaskFailed`  has different console.writeline content, i.e. line 334.
let's refine the log content latre.

**Review by herohua — APPROVED** [review]

**Pull request merged**